st2actions/st2actions/scheduler.py

```diff
@@ -56,26 +58,42 @@ def process(self, request):
             LOG.exception('Failed to find liveaction %s in the database.', request.id)
             raise
 
+        # Apply policies defined for the action.
+        for policy_db in Policy.query(resource_ref=liveaction_db.action):
+            policy = policies.get_driver(policy_db.ref,
+                                         policy_db.policy_type,
+                                         **policy_db.parameters)
+            liveaction_db = policy.apply(liveaction_db)
+
+        # Exit if the status of the request is no longer runnable.
+        # The status could have be changed by one of the policies.
+        if liveaction_db.status not in [action_constants.LIVEACTION_STATUS_REQUESTED,
+                                        action_constants.LIVEACTION_STATUS_SCHEDULED]:
+            LOG.info('%s is ignoring %s (id=%s) with "%s" status after policies are applied.',
```

Comment:
Would be interesting/useful to know which policy caused this scheduling push back.